app/uk/gov/hmrc/economiccrimelevyregistration/controllers/UkRevenueController.scala

```diff
@@ -47,6 +47,20 @@ class UkRevenueController @Inject() (
   }
 
   def onSubmit: Action[AnyContent] = (authorise andThen getRegistrationData).async { implicit request =>
-    ???
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors))),
+        meetsRevenueThreshold =>
+          eclRegistrationConnector
+            .upsertRegistration(request.registration.copy(meetsRevenueThreshold = Some(meetsRevenueThreshold)))
+            .flatMap { _ =>
+              if (meetsRevenueThreshold) {
+                Future.successful(Ok("Proceed"))
```

Comment:
I assume we are planning on implement the next page as part of this as well?

Reply:
Yes, once you are happy with what I've done so far I will move onto this page.